seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -0,0 +1,135 @@
+import json
+import mock
+
+from django.urls.base import reverse
+
+from seqr.models import Project
+from seqr.views.apis.anvil_workspace_api import anvil_workspace_page, create_project_from_workspace
+from seqr.views.utils.test_utils import AnvilAuthenticationTestCase
+
+WORKSPACE_NAMESPACE = 'my-seqr-billing'
+EXIST_WORKSPACE_NAME = 'anvil-project 1000 Genomes Demo'
+WORKSPACE_NAME = 'anvil project name'
+NEW_WORKSPACE_NAME = 'anvil new project'
+
+LOAD_SAMPLE_DATA = [
+    ["Family ID", "Individual ID", "Previous Individual ID", "Paternal ID", "Maternal ID", "Sex", "Affected Status",
+     "Notes", "familyNotes"],
+    ["1", "NA19675", "NA19675_1", "NA19678", "", "Female", "Affected", "A affected individual, test1-zsf", ""],
+    ["1", "NA19678", "", "", "", "Male", "Unaffected", "a individual note", ""],
+    ["21", "HG00735", "", "", "", "Female", "Unaffected", "", "a new family"]]
+
+BAD_SAMPLE_DATA = [["1", "NA19674", "NA19674_1", "NA19678", "NA19679", "Female", "Affected", "A affected individual, test1-zsf", ""]]
+
+REQUEST_BODY = {
+            'genomeVersion': '38',
+            'uploadedFileId': 'test_temp_file_id',
+            'description': 'A test project',
+            'agreeSeqrAccess': True,
+        }
+
+
+@mock.patch('seqr.views.utils.permissions_utils.user_get_workspace_access_level')
+class AnvilWorkspaceAPITest(AnvilAuthenticationTestCase):
+    fixtures = ['users', 'social_auth', '1kg_project']
+
+    @mock.patch('seqr.views.utils.permissions_utils.logger')
+    def test_anvil_workspace_page(self, mock_logger, mock_get_access_level):
+        # Requesting to load data for a non-existing project
+        url = reverse(anvil_workspace_page, args=[WORKSPACE_NAMESPACE, WORKSPACE_NAME])
+        self.check_require_login(url)
```

Comment:
All `check_require_login` does is check that if no one is logged in it returns a 302, but because the expected behavior on success is also 302, this isn't actually a sufficient test. You need to add a specific test before this that when a user is not logged in it redirects to the google login page. Moreover, you should use proper workspace names with the test data to be able to use the `check_collaborator_login` function to test for 403s instead of setting up your own permission failure test